kpops/components/base_components/models/from_section.py

```diff
@@ -38,19 +42,39 @@ class Config(DescConfig):
 
     @root_validator
     def extra_topic_role(cls, values: dict) -> dict:
-        """Ensure that cls.role is used correctly"""
-        is_extra_topic = values["type"] in (
-            InputTopicTypes.EXTRA,
-            InputTopicTypes.EXTRA_PATTERN,
-        )
-        if is_extra_topic and not values.get("role"):
-            raise ValueError(
-                "If you define an extra input component, extra input topic, or extra input pattern, you have to define a role."
-            )
-        if not is_extra_topic and values.get("role"):
-            raise ValueError(
-                "If you do not define an input component, input topic, or input pattern, the role is unnecessary."
-            )
+        """Ensure that cls.role is used correctly, assign type if needed"""
+        type_ = values.get("type", None)
+        role = values.get("role", None) is not None
+        # Assign type
+        match type_, role:
+            case None, False:
+                values["type"] = InputTopicTypes.INPUT
+                return values
+            case None, True:
+                values["type"] = InputTopicTypes.EXTRA
+                return values
+            case InputTopicTypes.PATTERN, False:
+                values["type"] = InputTopicTypes.INPUT_PATTERN
+                return values
+            case InputTopicTypes.PATTERN, True:
+                values["type"] = InputTopicTypes.EXTRA_PATTERN
+                return values
+            case _, _:
+                is_input_topic = values["type"] not in (
+                    InputTopicTypes.EXTRA,
+                    InputTopicTypes.EXTRA_PATTERN,
+                )
+        match is_input_topic, role:
+            case True, True:
+                raise ValueError(
+                    "If you define an extra input component, extra input topic, or extra input pattern, you have to define a role."
+                )
```

Comment:
shouldn't it be `True`, `False`?

Reply:
ah, I see you reversed the `is_extra_topic` to `is_input_topic`. may I ask why?

Reply:
No particular reason, I kept mishandling the cases and just ended up on that while trying to make it work. I guess it was easier to follow at the time. I will switch it around though, no need for the negation.